./gradlew clean build --profile produced spurious PMD errors

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':library:violations'.
> javax.script.ScriptException: Too many violations found, max is 0 but found 3
  /Users/judds/opensource/glide/library/src/main/java/com/bumptech/glide/load/engine/Engine.java
  ╔══════════╤═════════════════════╤══════════╤══════╤══════════════════════════════════════════════════════════════════════════════════════════════════════╗
  ║ Reporter │ Rule                │ Severity │ Line │ Message                                                                                              ║
  ╠══════════╪═════════════════════╪══════════╪══════╪══════════════════════════════════════════════════════════════════════════════════════════════════════╣
  ║ PMD      │ UnusedPrivateMethod │ WARN     │ 255  │ Avoid unused private methods such as 'loadFromCache(Key,boolean)'.                                   ║
  ║          │                     │          │      │                                                                                                      ║
  ║          │                     │          │      │                                                                                                      ║
  ║          │                     │          │      │ Best Practices https://pmd.github.io/pmd-6.0.0/pmd_rules_java_bestpractices.html#unusedprivatemethod ║
  ╟──────────┼─────────────────────┼──────────┼──────┼──────────────────────────────────────────────────────────────────────────────────────────────────────╢
  ║ PMD      │ UnusedPrivateMethod │ WARN     │ 243  │ Avoid unused private methods such as 'loadFromActiveResources(Key,boolean)'.                         ║
  ║          │                     │          │      │                                                                                                      ║
  ║          │                     │          │      │                                                                                                      ║
  ║          │                     │          │      │ Best Practices https://pmd.github.io/pmd-6.0.0/pmd_rules_java_bestpractices.html#unusedprivatemethod ║
  ╟──────────┼─────────────────────┼──────────┼──────┼──────────────────────────────────────────────────────────────────────────────────────────────────────╢
  ║ PMD      │ UnusedPrivateMethod │ WARN     │ 238  │ Avoid unused private methods such as 'logWithTimeAndKey(String,long,Key)'.                           ║
  ║          │                     │          │      │                                                                                                      ║
  ║          │                     │          │      │                                                                                                      ║
  ║          │                     │          │      │ Best Practices https://pmd.github.io/pmd-6.0.0/pmd_rules_java_bestpractices.html#unusedprivatemethod ║
  ╚══════════╧═════════════════════╧══════════╧══════╧══════════════════════════════════════════════════════════════════════════════════════════════════════╝
  
  Summary of /Users/judds/opensource/glide/library/src/main/java/com/bumptech/glide/load/engine/Engine.java
  ╔══════════╤══════╤══════╤═══════╤═══════╗
  ║ Reporter │ INFO │ WARN │ ERROR │ Total ║
  ╠══════════╪══════╪══════╪═══════╪═══════╣
  ║ PMD      │ 0    │ 3    │ 0     │ 3     ║
  ╟──────────┼──────┼──────┼───────┼───────╢
  ║          │ 0    │ 3    │ 0     │ 3     ║
  ╚══════════╧══════╧══════╧═══════╧═══════╝
  
  
  Summary
  ╔══════════╤══════╤══════╤═══════╤═══════╗
  ║ Reporter │ INFO │ WARN │ ERROR │ Total ║
  ╠══════════╪══════╪══════╪═══════╪═══════╣
  ║ PMD      │ 0    │ 3    │ 0     │ 3     ║
  ╟──────────┼──────┼──────┼───────┼───────╢
  ║          │ 0    │ 3    │ 0     │ 3     ║
  ╚══════════╧══════╧══════╧═══════╧═══════╝


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org
```

--- a/library/build.gradle
+++ b/library/build.gradle
@@ -97,6 +97,7 @@ afterEvaluate {
     }
 
     tasks.create('pmd', Pmd) {
+        dependsOn tasks.compileDebugJavaWithJavac
         targetJdk = TargetJdk.VERSION_1_7
 
         description 'Run pmd'
@@ -108,7 +109,7 @@ afterEvaluate {
         ruleSetFiles = files('pmd-ruleset.xml')
         source android.sourceSets.main.java.srcDirs
         classpath = files()
-        classpath += files("${project.buildDir}/intermediates/classes/debug/")
+        classpath += files(tasks.compileDebugJavaWithJavac.destinationDir)
         doFirst {
             classpath += classPathForQuality()
         }

--- a/library/build.gradle
+++ b/library/build.gradle
@@ -67,12 +67,12 @@ afterEvaluate {
     }
 
     tasks.create('findbugs', FindBugs) {
-        dependsOn tasks.assembleDebug
+        dependsOn tasks.compileDebugJavaWithJavac
 
         description 'Run findbugs'
         group 'verification'
 
-        classes = fileTree("${project.buildDir}/intermediates/classes/debug/")
+        classes = fileTree(tasks.compileDebugJavaWithJavac.destinationDir)
         source android.sourceSets.main.java.srcDirs
         classpath = files()
         doFirst {

--- a/library/src/main/java/com/bumptech/glide/load/engine/Engine.java
+++ b/library/src/main/java/com/bumptech/glide/load/engine/Engine.java
@@ -235,14 +235,10 @@ public class Engine implements EngineJobListener,
     return new LoadStatus(cb, engineJob);
   }
 
-  // TODO(#2776): Remove this suppression.
-  @SuppressWarnings("PMD.UnusedPrivateMethod")
   private static void logWithTimeAndKey(String log, long startTime, Key key) {
     Log.v(TAG, log + " in " + LogTime.getElapsedMillis(startTime) + "ms, key: " + key);
   }
 
-  // TODO(#2776): Remove this suppression.
-  @SuppressWarnings("PMD.UnusedPrivateMethod")
   @Nullable
   private EngineResource<?> loadFromActiveResources(Key key, boolean isMemoryCacheable) {
     if (!isMemoryCacheable) {
@@ -256,8 +252,6 @@ public class Engine implements EngineJobListener,
     return active;
   }
 
-  // TODO(#2776): Remove this suppression.
-  @SuppressWarnings("PMD.UnusedPrivateMethod")
   private EngineResource<?> loadFromCache(Key key, boolean isMemoryCacheable) {
     if (!isMemoryCacheable) {
       return null;

